File: src/main/java/com/elchologamer/userlogin/util/FastLoginHook.java
package com.elchologamer.userlogin.util;

import com.elchologamer.userlogin.UserLogin;
import com.elchologamer.userlogin.api.types.AuthType;
import com.elchologamer.userlogin.command.AuthCommand;
import com.elchologamer.userlogin.ULPlayer;
import com.github.games647.fastlogin.bukkit.FastLoginBukkit;
import com.github.games647.fastlogin.core.hooks.AuthPlugin;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.UUID;

public class FastLoginHook implements AuthPlugin<Player> {

    private final UserLogin plugin = UserLogin.getPlugin();

    public void register() {
        FastLoginBukkit fastLogin = JavaPlugin.getPlugin(FastLoginBukkit.class);
        fastLogin.getCore().setAuthPluginHook(new FastLoginHook());
    }

    @Override
    public boolean forceLogin(Player player) {
        ULPlayer ulPlayer = plugin.getPlayer(player);
        if (ulPlayer.isLoggedIn()) return false;

        plugin.getServer().getScheduler().runTask(
                plugin,
                () -> AuthCommand.login(ulPlayer, AuthType.LOGIN)
        );
        return true;
    }

    @Override
    public boolean forceRegister(Player player, String password) {
        try {
            plugin.getDB().createPassword(player.getUniqueId(), password);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    @Override
    public boolean isRegistered(String playerName) {
        UUID uuid = Utils.fetchPlayerUUID(playerName);
        if (uuid == null) return false;

        return plugin.getDB().isRegistered(uuid);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/util/FastLoginHook.kt
package com.elchologamer.userlogin.util

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import com.elchologamer.userlogin.api.types.AuthType
import com.github.games647.fastlogin.bukkit.FastLoginBukkit
import com.github.games647.fastlogin.core.hooks.AuthPlugin
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin

object FastLoginHook : AuthPlugin<Player> {
    fun register() {
        val fastLogin = JavaPlugin.getPlugin(FastLoginBukkit::class.java)
        fastLogin.core.authPluginHook = this
    }

    override fun forceLogin(player: Player): Boolean {
        val ulPlayer = plugin.getPlayer(player)
        if (ulPlayer.loggedIn) return false

        plugin.server.scheduler.runTask(
            plugin,
            Runnable { ulPlayer.onAuthenticate(AuthType.LOGIN) }
        )
        return true
    }

    override fun forceRegister(player: Player, password: String): Boolean {
        return try {
            plugin.db.createPassword(player.uniqueId, password)
            true
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    override fun isRegistered(playerName: String): Boolean {
        val uuid = Utils.fetchPlayerUUID(playerName) ?: return false
        return plugin.db.isRegistered(uuid)
    }
}